### PR TITLE
chore: enable more ts test

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
1. Part of https://github.com/rolldown/rolldown/issues/1476
2. Error tracking here, https://github.com/rolldown/rolldown/issues/1513

### CODE REVIEW

File: crates/rolldown/tests/esbuild/bundler_ts/ts_abstract_class_field_use_define/artifacts.snap [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+```js
+
+//#region entry.ts
+const keepThisToo = Symbol('keepThisToo');
+class Foo {
+	keepThis;
+	[keepThisToo];
+}
+(() => new Foo())();
+
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/bundler_ts/ts_abstract_class_field_use_define/artifacts.snap:19]:
```
TestTSAbstractClassFieldUseDefine
---------- /out.js ----------
const keepThisToo = Symbol("keepThisToo");
class Foo {
  keepThis;
  [keepThisToo];
}
(() => new Foo())();

```

File: crates/rolldown/tests/esbuild/bundler_ts/ts_import_cts/artifacts.snap [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+
+//#region required.cjs
+var require_required = __commonJSMin((exports, module) => {
+	console.log('works');
+});
+
+//#endregion
+//#region entry.ts
+require_required();
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/bundler_ts/ts_import_cts/artifacts.snap:20]:
```
TestTSImportCTS
---------- /out.js ----------
// required.cjs
var require_required = __commonJS({
  "required.cjs"() {
    console.log("works");
  }
});

// entry.ts
require_required();

```

File: crates/rolldown/tests/esbuild/bundler_ts/ts_declare_function/artifacts.snap [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+input_file: crates/rolldown/tests/esbuild/bundler_ts/ts_declare_function
+---
+# Assets
+
+## entry_ts.mjs
+
+```js
+
+//#region entry.ts
+let foo = bar();
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/bundler_ts/ts_declare_function/artifacts.snap:13]:
```
TestTSDeclareFunction
---------- /out.js ----------
// entry.ts
var foo = bar();
```

File: crates/rolldown/tests/esbuild/bundler_ts/ts_import_empty_namespace/artifacts.snap [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+---
+# Assets
+
+## entry_ts.mjs
+
+```js
+
+//#region entry.ts
+function foo() {}
+foo();
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/bundler_ts/ts_import_empty_namespace/artifacts.snap:14]:
```
TestTSImportEmptyNamespace
---------- /out.js ----------
// entry.ts
function foo() {
}
foo();
```

File: crates/rolldown/tests/esbuild/bundler_ts/ts_declare_enum/artifacts.snap [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+expression: content
+input_file: crates/rolldown/tests/esbuild/bundler_ts/ts_declare_enum
+---
+# Assets
+
+## entry_ts.mjs
+
+```js
+
+//#region entry.ts
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/bundler_ts/ts_declare_enum/artifacts.snap:12]:
```
TestTSDeclareEnum
---------- /out.js ----------
// entry.ts
var foo = bar();
```

File: crates/rolldown/tests/esbuild/bundler_ts/ts_declare_class/artifacts.snap [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+---
+# Assets
+
+## entry_ts.mjs
+
+```js
+
+//#region entry.ts
+let foo = bar();
+
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/bundler_ts/ts_declare_class/artifacts.snap:14]:
```
TestTSDeclareClass
---------- /out.js ----------
// entry.ts
var foo = bar();
```

File: crates/rolldown/tests/esbuild/bundler_ts/ts_declare_const_enum/artifacts.snap [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+input_file: crates/rolldown/tests/esbuild/bundler_ts/ts_declare_const_enum
+---
+# Assets
+
+## entry_ts.mjs
+
+```js
+
+//#region entry.ts
+let foo = bar();
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/bundler_ts/ts_declare_const_enum/artifacts.snap:13]: TestTSDeclareConstEnum
---------- /out.js ----------
// entry.ts
var foo = bar();

File: crates/rolldown/tests/esbuild/bundler_ts/ts_declare_namespace/artifacts.snap [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+input_file: crates/rolldown/tests/esbuild/bundler_ts/ts_declare_namespace
+---
+# Assets
+
+## entry_ts.mjs
+
+```js
+
+//#region entry.ts
+let foo = bar();
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/bundler_ts/ts_declare_namespace/artifacts.snap:13]:
```js
TestTSDeclareNamespace
---------- /out.js ----------
// entry.ts
var foo = bar();

```

File: crates/rolldown/tests/esbuild/bundler_ts/ts_import_type_only_file/artifacts.snap [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+input_file: crates/rolldown/tests/esbuild/bundler_ts/ts_import_type_only_file
+---
+# Assets
+
+## entry_ts.mjs
+
+```js
+
+//#region entry.ts
+let foo = bar();
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/bundler_ts/ts_import_type_only_file/artifacts.snap:13]: TestTSImportTypeOnlyFile
---------- /out.js ----------
// entry.ts
var foo = bar();

**Review by hyf0 — APPROVED** [review]

**Pull request merged**